src/tables/colr.rs

```diff
@@ -47,16 +189,520 @@ impl FromData for LayerRecord {
+            color,
+        })
+    }
+}
+
+#[cfg(feature = "variable-fonts")]
+impl VarColorLine<'_> {
+    // TODO: Color stops should be sorted, but hard to do without allocations
+    fn get(
+        &self,
+        palette: u16,
+        index: u16,
+        #[cfg(feature = "variable-fonts")] variation_data: VariationData,
+        #[cfg(feature = "variable-fonts")] coordinates: &[NormalizedCoordinate],
+    ) -> Option<ColorStop> {
+        let info = self.colors.get(index)?;
+
+        let mut color = if info.palette_index == u16::MAX {
+            self.foreground_color
+        } else {
+            self.palettes.get(palette, info.palette_index)?
+        };
+
+        let deltas = variation_data.read_deltas::<2>(info.var_index_base, coordinates);
+        let stop_offset = info.stop_offset.apply_float_delta(deltas[0]);
+        color.apply_alpha(info.alpha.apply_float_delta(deltas[1]));
+
+        Some(ColorStop { stop_offset, color })
+    }
+}
+
+#[cfg(feature = "variable-fonts")]
+#[derive(Clone)]
+struct VarColorLine<'a> {
+    extend: GradientExtend,
+    colors: LazyArray16<'a, VarColorStopRaw>,
+    palettes: cpal::Table<'a>,
+    foreground_color: RgbaColor,
+}
+
+#[derive(Clone)]
+enum ColorLine<'a> {
+    #[cfg(feature = "variable-fonts")]
+    VarColorLine(VarColorLine<'a>),
+    NonVarColorLine(NonVarColorLine<'a>),
+}
+
+/// A [gradient extend](
+/// https://learn.microsoft.com/en-us/typography/opentype/spec/colr#baseglyphlist-layerlist-and-cliplist).
+#[derive(Clone, Copy, Debug)]
+pub struct ColorStop {
+    /// The offset of the color stop.
+    pub stop_offset: f32,
+    /// The color of the color stop.
+    pub color: RgbaColor,
+}
+
+/// A [linear gradient](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#formats-4-and-5-paintlineargradient-paintvarlineargradient)
+#[derive(Clone)]
+pub struct LinearGradient<'a> {
+    /// The `x0` value of the linear gradient.
+    pub x0: f32,
+    /// The `y0` value of the linear gradient.
+    pub y0: f32,
+    /// The `x1` value of the linear gradient.
+    pub x1: f32,
+    /// The `y1` value of the linear gradient.
+    pub y1: f32,
+    /// The `x2` value of the linear gradient.
+    pub x2: f32,
+    /// The `y2` value of the linear gradient.
+    pub y2: f32,
+    /// The extend of the gradient.
+    pub extend: GradientExtend,
+    #[cfg(feature = "variable-fonts")]
+    variation_data: VariationData<'a>,
+    color_line: ColorLine<'a>,
+}
+
+impl<'a> core::fmt::Debug for LinearGradient<'a> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("LinearGradient")
+            .field("x0", &self.x0)
+            .field("y0", &self.y0)
+            .field("x1", &self.x1)
+            .field("y1", &self.y1)
+            .field("x2", &self.x2)
+            .field("y2", &self.y2)
+            .field("extend", &self.extend)
+            // .field("stops", &self.stops(0))
```

Comment:
?

Reply:
Well, the problem is right now you need to pass the variation coordinates directly to the `stops` function... But I gues I can just add empty coordinates for the debug function.

Reply:
Added empty ones.